client/core/core.go

```diff
@@ -7635,12 +7665,29 @@ const (
 // route. If temporary is provided and true, the c.listen(dc) goroutine is not
 // started so that associated trades are not processed and no incoming requests
 // and notifications are handled. A temporary dexConnection may be used to
-// inspect the config response or check if a (paid) HD account exists with a DEX.
+// inspect the config response or check if a (paid) HD account exists with a
+// DEX.  If we fail to connect, there would be no retries.
 func (c *Core) connectDEX(acctInfo *db.AccountInfo) (*dexConnection, error) {
 	return c.connectDEXWithFlag(acctInfo, 0)
 }
 
 func (c *Core) connectDEXWithFlag(acctInfo *db.AccountInfo, flag connectDEXFlag) (*dexConnection, error) {
+	dc, err := c.newDEXConnection(acctInfo, flag)
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.startDexConnection(acctInfo, dc)
+	if err != nil {
+		dc.connMaster.Disconnect() // stop any retry loop for this new connection.
+		return nil, err
```

Comment:
This is right for 90% of the use cases, but it breaks critical functionality of `initialize`.  The `initialize` method will have to work differently (just logging errors from `c.startDexConnection` but not stopping the conn master because the reconnect loop is there for the initial connection, not just reestablishing a dropped connection.